### PR TITLE
Remove the Forward and Share buttons for location messages only

### PR DESCRIPTION
**Pull request by andybalaam**
---
Here's what your changelog entry will look like:

## ✨ Features
 * Remove the Forward and Share buttons for location messages only ([\#7423](https://github.com/matrix-org/matrix-react-sdk/pull/7423)). Contributed by @andybalaam.

Preview: https://61c1cd41631e564a262011a5--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/context_menus/MessageContextMenu.tsx [hunk 1]
```diff
@@ -486,3 +500,14 @@ export default class MessageContextMenu extends React.Component<IProps, IState>
         );
     }
 }
+
+function isLocationEvent(event: MatrixEvent): boolean {
+    const eventType = event.getType();
+    return (
+        LOCATION_EVENT_TYPE.matches(eventType) ||
+        (
+            eventType === "m.room.message" &&
```
Comment (t3chguy) [inline src/components/views/context_menus/MessageContextMenu.tsx:509]: EventType.RoomMessage

Comment (andybalaam) [reply src/components/views/context_menus/MessageContextMenu.tsx]: Fixed, thanks.

**Comment by t3chguy** [comment]
Personally think it could be useful to forward/share locations, what's the rationale here?

File: src/components/views/context_menus/MessageContextMenu.tsx [hunk 2]
```diff
@@ -92,6 +95,8 @@ export default class MessageContextMenu extends React.Component<IProps, IState>
     state = {
         canRedact: false,
         canPin: false,
+        canForward: false,
+        canShare: false,
```
[the remark this answers is not in the document]
Comment (andybalaam) [reply src/components/views/context_menus/MessageContextMenu.tsx:99]: Thanks - I thought that, but was copying some that were there.  I'll fix it.

[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/context_menus/MessageContextMenu.tsx:99]: The others can vary due to permissions

**Comment by andybalaam** [comment]
> Personally think it could be useful to forward/share locations, what's the rationale here?

The designs are being as cautious as possible about privacy issues, and that is the motivation here.

File: src/components/views/context_menus/MessageContextMenu.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (andybalaam) [reply src/components/views/context_menus/MessageContextMenu.tsx]: Fixed in db67c2c54b

[the remark this answers is not in the document]
Comment (andybalaam) [reply src/components/views/context_menus/MessageContextMenu.tsx]: > The others can vary due to permissions changing whilst the thing is rendered

Ah, got it - thanks.

**Comment by t3chguy** [comment]
FTR WhatsApp allows users to forward a location message, so users will likely be expecting the same level of functionality

**Review by t3chguy — APPROVED** [review]

**Comment by andybalaam** [comment]
> FTR WhatsApp allows users to forward a location message, so users will likely be expecting the same level of functionality

cc @gaelledel @jakewb-b

**Pull request merged**